Show header for 'lotus route' command only one time

davydovanton opened the pull request:
This PR do two things:

Firstly, it replaces conditional with polymorphism in `RoutesInspector#to_s` and it closes one TODO task.
Secondly, it fixes displaying header for `lotus routes` command(#82). Now this header shows only one time.

jodosha commented:
@davydovanton Hey, thanks for this PR. It changes too much for a "print an extra line for the inspector" fix. What I don't like is that it increases the API of Router with an extra method for inspection.

Can we just have this fix (with a test) for now? Thank you very much.

davydovanton commented:
@jodosha the base problem in recursion calling `to_s` method. Therefore If you add some title to `#to_s` this method displays this title repeatedly.
I can revert refactoring changes and create private method only for displaying raw text about routes (without any titles or headers).
If you not oppose to such a solution, I can fix it.

P.S.: so, what about refactoring? I can open separate PR only with this.

jodosha commented:
@davydovanton Please just fix that problem and leave the rest as it is. It's not important, really. We want to focus on new features, rather than on these small details :wink:

davydovanton commented:
@jodosha I reverted refactoring changes and I replaced recursive call to separate method. Now when you call `lotus routes` command you'll see something like this:
```
                Name Method     Path                           Action

                     GET, HEAD  /fakeroute1                    Fake::Index
                     GET, HEAD  /fakeroute3                    Fake::Index
                     GET, HEAD  /fakeroute2                    Fake::Index
                     GET, HEAD  /fakeroute4                    Fake::Index
                     GET, HEAD  /api/posts                     Posts::Index
                     GET, HEAD  /api/posts_all                 Posts::Index
                     GET, HEAD  /api/second_mount/comments     Comments::Index
                     GET, HEAD  /api/second_mount/comments1    Comments::Index
                     GET, HEAD  /api/second_mount/comments2    Comments::Index
```

jodosha commented:
@davydovanton Thanks for this update, but there aren't tests here to prove this change. Can you please add them? Thank you! :+1:

davydovanton commented:
@jodosha done :)

jodosha commented:
@davydovanton Thank you :+1:

davydovanton commented:
Thanks for review :+1:
